invisible_cities/io/hist_io_test.py

```diff
@@ -0,0 +1,105 @@
+    save_histomanager_to_file(histogram_manager, file_out, group=group)
+
+    with tb.open_file(file_out, "r") as h5in:
+        file_group = getattr(h5in.root, group)
+        for histogram in list_of_histograms:
+            histoname    = histogram.title
+            saved_labels = [str(label)[2:-1].replace('\\\\', '\\') for label in getattr(file_group, histoname + "_labels")[:]]
+
+            assert histoname in file_group
+            assert np.all     (a == b for a, b in zip(histogram.bins, getattr(file_group, histoname + "_bins")[:]))
+            assert np.allclose(histogram.data     , getattr(file_group, histoname              )[:])
+            assert np.allclose(histogram.errors   , getattr(file_group, histoname + "_errors"  )[:])
+            assert np.allclose(histogram.out_range, getattr(file_group, histoname + "_outRange")[:])
+            assert             histogram.labels  == saved_labels
+
+
+@settings(suppress_health_check=(HealthCheck.too_slow,))
+@given   (histograms_lists(), text(characters, min_size=1))
+def test_save_histomanager_to_file_append_mode(output_tmpdir, histogram_list, group):
+    args, list_of_histograms = histogram_list
+    histogram_manager        = HistoManager(list_of_histograms[:1])
+
+    file_out          = os.path.join(output_tmpdir, 'test_save_histogram_manager.h5')
+    save_histomanager_to_file       (histogram_manager, file_out, mode='w', group=group)
+    histogram_manager = HistoManager(list_of_histograms[1:])
+    save_histomanager_to_file       (histogram_manager, file_out, mode='a', group=group)
+
+    with tb.open_file(file_out, "r") as h5in:
+        file_group = getattr(h5in.root, group)
+        for histogram in list_of_histograms:
+            histoname    = histogram.title
+            saved_labels = [str(label)[2:-1].replace('\\\\', '\\') for label in getattr(file_group, histoname + "_labels")[:]]
+
+            assert histoname in file_group
+            assert np.all     (a == b for a, b in zip(histogram.bins, getattr(file_group, histoname + "_bins")[:]))
+            assert np.allclose(histogram.data     , getattr(file_group, histoname              )[:])
+            assert np.allclose(histogram.errors   , getattr(file_group, histoname + "_errors"  )[:])
+            assert np.allclose(histogram.out_range, getattr(file_group, histoname + "_outRange")[:])
+            assert             histogram.labels  == saved_labels
+
+
+@settings(suppress_health_check=(HealthCheck.too_slow,))
+@given   (histograms_lists(), text(characters, min_size=1), text(characters, min_size=1, max_size=1))
+def test_save_histomanager_to_file_raises_ValueError(output_tmpdir, histogram_list, group, write_mode):
+    args, list_of_histograms = histogram_list
+    histogram_manager        = HistoManager(list_of_histograms)
+
+    file_out   = os.path.join      (output_tmpdir, 'test_save_histogram_manager.h5')
+    write_mode = write_mode.replace('w','')
+    write_mode = write_mode.replace('a','')
```

Comment:
It is better to exclude `"w"` and `"a"` from the input. Try `text(characters, min_size=1, max_size=1).filter(lambda x: x not in ["w", "a"])`

Reply:
I meant, add the `filter` to the input and remove the `replace` calls here :)